### PR TITLE
fix(glue): Make Glue Database locationUri optional.

### LINKED ISSUE #5268: Glue database: is locationUri required?
### The Question

Hi!

I tried creating glue database and noticed that if I don't specify locationUri, CDK creates one for me by creating an S3 bucket.
Here's the [source code](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-glue/lib/database.ts#L97)

Just wondering, was that that intentional? Location URI is not a required parameter in Cloudformation - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-database-databaseinput.html#cfn-glue-database-databaseinput-locationuri .

### Environment

  - **CDK CLI Version:** 1.18.0 
  - **Module Version:** 1.18.0
  - **OS:** GNU/Linux
  - **Language:** TypeScript

### PR DESCRIPTION
**Pull request by melgenek**
Do not create an s3 bucket by default.
Closes #5268

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-glue/lib/database.ts [hunk 1]
```diff
@@ -83,27 +82,29 @@ export class Database extends Resource implements IDatabase {
...
     }
 
     this.catalogId = Stack.of(this).account;
     const resource = new CfnDatabase(this, 'Resource', {
       catalogId: this.catalogId,
-      databaseInput: {
-        name: this.physicalName,
-        locationUri: this.locationUri
-      }
+      databaseInput: databaseInput
```
Comment (skinny85) [inline packages/@aws-cdk/aws-glue/lib/database.ts]: I believe you need to change this to just `databaseInput` to silence the linter that's making the build fail.

File: packages/@aws-cdk/aws-glue/lib/database.ts [hunk 2]
```diff
@@ -83,27 +82,29 @@ export class Database extends Resource implements IDatabase {
...
     }
 
     this.catalogId = Stack.of(this).account;
     const resource = new CfnDatabase(this, 'Resource', {
       catalogId: this.catalogId,
-      databaseInput: {
-        name: this.physicalName,
-        locationUri: this.locationUri
-      }
+      databaseInput
```
Comment (skinny85) [inline packages/@aws-cdk/aws-glue/lib/database.ts]: Trailing comma please :).

File: packages/@aws-cdk/aws-glue/lib/database.ts [hunk 3]
```diff
@@ -83,27 +82,29 @@ export class Database extends Resource implements IDatabase {
...
   constructor(scope: Construct, id: string, props: DatabaseProps) {
     super(scope, id, {
       physicalName: props.databaseName,
     });
 
+    let databaseInput: CfnDatabase.DatabaseInputProperty = {
+      name: props.databaseName,
+    };
+
     if (props.locationUri) {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-glue/lib/database.ts]: This should probably be `props.locationUri !== undefined` instead, to correctly handle the empty location case (not sure if it's valid, but probably should be possible).

Comment (melgenek) [reply packages/@aws-cdk/aws-glue/lib/database.ts]: The idea is actually to exclude `locationUri` if it is an empty string.
Aws does not allow empty strings there. See https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-databases.html 
```
LocationUri – Uniform resource identifier (uri), not less than 1 or more than 1024 bytes long...
```
So we have to exclude uri from the `databaseInput`. This is what `if (props.locationUri)` (empty string is false in js). But I can make it less magical by writing smth like
```
if(props.locationUri !== undefined && props.locationUri.length >= 1) {...}
```

Comment (skinny85) [reply packages/@aws-cdk/aws-glue/lib/database.ts]: Then we should fail with an exception if the passed `locationUri` is empty. Would you mind doing that (+ a unit test for this behavior)?

Comment (melgenek) [reply packages/@aws-cdk/aws-glue/lib/database.ts]: @skinny85 Showing an exception makes sense. Updated the pr, and removed the mention about the s3 from the readme file.

**Pull request merged**